src/js/actions/documents.js

```diff
@@ -273,27 +273,38 @@ define(function (require, exports) {
      * Initialize document and layer state, emitting DOCUMENT_UPDATED events, for
      * all the inactive documents.
      *
-     * @param {number} currentIndex
+     * @param {number} currentDocID
      * @param {number} docCount
      * @return {Promise}
      */
-    var initInactiveDocuments = function (currentIndex, docCount) {
-        var otherDocPromises = _.range(1, docCount + 1)
-            .filter(function (index) {
-                return index !== currentIndex;
-            })
+    var initInactiveDocuments = function (currentDocID, docCount) {
+        var docReferences = _.range(1, docCount + 1)
             .map(function (index) {
-                var indexRef = documentLib.referenceBy.index(index);
-
-                // Only load essential properties for inactive documents
-                return _getInactiveDocumentByRef(indexRef)
-                    .bind(this)
-                    .then(function (document) {
-                        this.dispatch(events.document.DOCUMENT_UPDATED, {
-                            document: document
-                        });
+                return documentLib.referenceBy.index(index);
+            }),
+            otherDocPromises = descriptor.batchGetProperty(docReferences, "documentID")
+                .bind(this)
+                .then(function (documentIDs) {
+                    return documentIDs.filter(function (documentID) {
+                        return documentID !== currentDocID;
                     });
-            }, this);
+                })
+                .map(function (documentID) {
+                    var docRef = documentLib.referenceBy.id(documentID);
+                   
+                    // Only load essential properties for inactive documents
+                    return _getInactiveDocumentByRef(docRef)
+                        .bind(this)
+                        .then(function (document) {
+                            if (document.documentID === currentDocID) {
```

Comment:
Just get rid of this check. I don't want to clutter the code with checks for every possible crazy thing Photoshop could possibly do, and there's no reason why you're `filter` step above wouldn't work reliably.